corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/utils.js

```diff
@@ -85,18 +85,30 @@ hqDefine("cloudcare/js/formplayer/menus/utils", function () {
         FormplayerFrontend.regions.getRegion('breadcrumb').show(breadcrumbView);
     };
 
-    var showFormMenu = function (langs, enableLanguageMenu) {
+    var showFormMenu = function (langs) {
         var langModels,
             langCollection;
 
         FormplayerFrontend.regions.addRegions({
             formMenu: "#form-menu",
         });
-        if (langs && enableLanguageMenu) {
-            langModels = _.map(langs, function (lang) {
-                return {
-                    lang: lang,
-                };
+        if (langs) {
+            $.getJSON('/langcodes/langs.json', function (mapping) {
+                langModels = _.map(langs, function (lang) {
+                    var matchingLanguage = mapping.find(function (language) {
+                        return language.code === lang;
+                    });
+                    return {
+                        lang_code: lang,
+                        lang_label: matchingLanguage ? matchingLanguage.name : lang,
+                    };
+                });
+
+                langCollection = new Backbone.Collection(langModels);
+                var formMenuView = views.FormMenuView({
+                    collection: langCollection,
+                });
+                FormplayerFrontend.regions.getRegion('formMenu').show(formMenuView);
```

Comment:
I see this stuff is duplicated with the code below - is it necessary in both places?  Does the `$.getJSON` call mean this block happens asynchronously after the initial render or something?

Reply:
Yes, this code is executed asynchronously and therefore duplicated code was added  within the `if` check to execute `FormplayerFrontend.regions.getRegion('formMenu').show(formMenuView)` after result is fetched.

Made the call synchronous and removed code duplication.

Looking for a way to make it available in context as per https://github.com/dimagi/commcare-hq/pull/33158#discussion_r1250536328 which will remove this `langs.json` fetching call as well here and instead will be read from the context data.

Reply:
Update - Removed the  async call to get the language mapping and instead passing it in page context.
https://github.com/dimagi/commcare-hq/pull/33158/commits/fd47c981452221abe96b2a63591698611a2cfaed